addon/src/mocks/request-wrapper.js

```diff
@@ -49,12 +49,13 @@ export default class RequestWrapper {
   }
 
   /**
-   * Sort the handlers by those with query params first
-   *
+   * Sort the handlers by those with queryParams first, then matchArgs second, then anything else
    */
   getHandlers() {
-    return this.handlers.sort(
-      (a, b) => b.hasQueryParams() - a.hasQueryParams(),
+    return [...this.handlers].sort(
+      (a, b) =>
+        b.hasQueryParams() - a.hasQueryParams() ||
+        Boolean(b.matchArgs) - Boolean(a.matchArgs),
```

Comment:
allows for more gauranteed ordering of which handler matches a request, regardless of which order they are added
```ts
mockCreate('thing') // third
mockCreate('thing').withParams(...) // first
mockCreate('thing').matches(...) // second
```